src/services/proposals.ts

```diff
@@ -30,3 +30,70 @@ export async function createProposalOnDb(mySql: ServerlessMysql, data: CreatePro
 
   return { proposalId };
 }
+
+export interface IProposalSqlRow {
+  proposal: string,
+  hashed_auth_password: string,
+  auth_password_salt: string,
+  version: number,
+  partial_tx: string,
+  signatures: string,
+  history: string,
+  created_at: string,
+  updated_at: string,
+}
+
+export interface IGetProposalFromDb {
+  id: string,
+  hashedAuthPassword: string,
+  authPasswordSalt: string,
+  partialTx: string,
+  signatures: string,
+  timestamp: string,
+  version: number,
+  history: {partialTx: string, timestamp: string}[]
+}
+
+export async function getProposalFromDb(mySql: ServerlessMysql, proposalId: string) {
+  const sqlRows: IProposalSqlRow[] = await mySql.query(
+    `SELECT proposal
+                 , hashed_auth_password
+                 , auth_password_salt
+                 , version
+                 , partial_tx
+                 , signatures
+                 , history
+                 , updated_at 
+            FROM proposals
+            WHERE proposal = ?`,
+    [proposalId],
+  );
+
+  if (sqlRows.length === 0) {
+    return null;
+  }
+
+  const sqlRow = sqlRows[0];
+  const jsObject: IGetProposalFromDb = {
+    id: sqlRow.proposal,
+    hashedAuthPassword: sqlRow.hashed_auth_password,
+    authPasswordSalt: sqlRow.auth_password_salt,
+    partialTx: sqlRow.partial_tx,
+    signatures: sqlRow.signatures,
+    timestamp: sqlRow.updated_at.toString(),
+    version: sqlRow.version,
+    history: [],
+  };
+
+  if (sqlRow.history?.length) {
+    try {
+      jsObject.history = JSON.parse(sqlRow.history);
+    } catch (e) {
```

Comment:
The history being corrupted on the database seems a critical error to me, right?

We should remember to add this lambda to the list of logGroups the log-manager is watching:

https://github.com/HathorNetwork/log-manager/blob/9e660fa7c30715783236ad2affe973ee0d2c20aa/serverless.yml#L130

Reply:
I think we should add a method to send alerts to the sqs instead of using the log-manager.
Since this project will run on AWS serverless and adding the role to allow access to the SQS will be easier.

Reply:
Both the `log-manager` and `alerts-manager` integrations are being documented on the [infrastructure design](https://github.com/HathorNetwork/internal-issues/issues/157) and will be implemented on future PRs :+1: 

Thanks for the heads up!

Reply:
I just did exactly this on the wallet-service, you can use this PR as reference:

https://github.com/HathorNetwork/hathor-wallet-service/pull/402